### PR TITLE
Some fixes from scanbuild and coverity

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Some things are just silencing the analysis tools.

### CODE REVIEW

File: src/lib/parser.c [hunk 1]
```diff
@@ -249,7 +257,7 @@ bool handle_attr_event(yaml_event_t *event,
         return push_handler(state, on_seq_scalar_event);
     case YAML_SEQUENCE_END_EVENT:
         /* XXX we know that sequences never come first so the previous state (map) has the key */
-        assert(state->s);
+        assert_analyzer(state->s);
```
Comment (williamcroberts) [inline src/lib/parser.c:260]: Where is this one coming from, is --debug performed at configure time? The assert() goes away if NDEBUG is defined, and --debug control's that. Where is this getting generated?

Comment (AndreasFuchsSIT) [reply src/lib/parser.c:260]: you are correct, scan-build runs ok with --enable-debug... hmpf...

**Comment by AndreasFuchsSIT** [comment]
P.S. coverity hates twist like crazy; complaining about 4 resource leaks false-positive because of the pointer-voodoo... ;-)

**Comment by williamcroberts** [comment]
> P.S. coverity hates twist like crazy; complaining about 4 resource leaks false-positive because of the pointer-voodoo... ;-)

Yeah it does!

File: src/lib/parser.c (hunk 1 above)
Comment (williamcroberts) [reply src/lib/parser.c:260]: Iv'e been burned so many times going, WTH... and then going oh yeah its not a debug build so the asserts are gone.

**Pull request merged**